proto/osmosis/concentrated-liquidity/tx.proto

```diff
@@ -27,20 +27,16 @@ message MsgCreatePosition {
   string sender = 2 [ (gogoproto.moretags) = "yaml:\"sender\"" ];
   int64 lower_tick = 3 [ (gogoproto.moretags) = "yaml:\"lower_tick\"" ];
   int64 upper_tick = 4 [ (gogoproto.moretags) = "yaml:\"upper_tick\"" ];
-  cosmos.base.v1beta1.Coin token_desired0 = 5 [
-    (gogoproto.moretags) = "yaml:\"token_desired0\"",
-    (gogoproto.nullable) = false
-  ];
-  cosmos.base.v1beta1.Coin token_desired1 = 6 [
-    (gogoproto.moretags) = "yaml:\"token_desired1\"",
-    (gogoproto.nullable) = false
+  repeated cosmos.base.v1beta1.Coin tokens_provided = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
```

Comment:
Why do we have these as coins? Why can't we assume that if a user attempts to create a position in a pool where pool has denom A as token 0 and B as token 1, then the denoms are implied?

Reply:
discussed offline, sg